### PR TITLE
Add support for sslmode=verify-full for mysql and pg

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
This means full verification of rootCA and hostname. The value for the hostname to check against is either 'sslhost' or 'host', 'sslhost' takes precedence when it is not empty.

### What does this PR do?
- _What's changed? Why were these changes made?_

  + Adds support for  sslmode=verify-full for mysql and pg
  + Adds `sslhost` option that can override the `host` for certificate hostname verification
  + Add integration tests

- _How should the reviewer approach this PR, especially if manual tests are required?_

  The entrypoint to the feature is `internal/plugin/connectors/tcp/ssl/ssl.go`. This file contains the TLS implementation for both mysql and pg.

- _Are there relevant screenshots you can add to the PR description?_
  N/A
### What ticket does this PR close?
Connected to https://github.com/cyberark/secretless-broker/issues/548

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] This PR does not require updating any documentation, or
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs. https://github.com/cyberark/secretless-docs/issues/255

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+- MySQL and PostgreSQL connectors support SSL host name verification with `verify-full`
+  SSL mode. Also adds optional `sslhost` configuration parameter that is compared to the
+  server's certificate SAN. [#1249](https://github.com/cyberark/secretless-broker/pull/1249)
```
Comment (izgeri) [inline CHANGELOG.md:12]: can we please refer to the issue it's resolving rather than the PR? usually the PRs have less contextual info

Comment (doodlesbykumbi) [reply CHANGELOG.md]: Done

File: internal/plugin/connectors/tcp/ssl/ssl.go [hunk 2]
```diff
@@ -51,18 +51,29 @@ func NewDbSSLMode(o options, requireCanVerifyCAOnly bool) (DbSSLMode, error) {
...
-	//case "verify-full":
-	//	sslMode.ServerName = o["host"]
+
+	case "verify-full":
+		// Use stdlib's verification
+		sslMode.InsecureSkipVerify = false
+		sslMode.VerifyCaOnly = false
+
+		// 'sslhost', when not empty, takes precedence over 'host'
+		if o["sslhost"] != "" {
```
Comment (BradleyBoutcher) [inline internal/plugin/connectors/tcp/ssl/ssl.go:69]: I'm not sure if this should be an empty string check or a length check. In either case, my follow-up PR will use a structured representation instead, so this shouldn't be an issue.

File: internal/plugin/connectors/tcp/mysql/connection_details.go [hunk 3]
```diff
@@ -2,57 +2,73 @@ package mysql
...
-	if credentials["password"] != nil {
+	if len(credentials["password"]) > 0 {
 		connDetails.Password = string(credentials["password"])
 	}
 
-	// Make sure that we process the SSL mode arg even if it's not specified
-	// otherwise it will get ignored
-	if _, ok := credentials["sslmode"]; !ok {
-		credentials["sslmode"] = []byte("")
+	for _, sslOption := range sslOptions {
```
Comment (BradleyBoutcher) [inline internal/plugin/connectors/tcp/mysql/connection_details.go:57]: Below you use a shorter method for this:
`connectionDetails.SSLOptions[sslOption] = string(options[sslOption])`
I think you can use that here too

**Pull request merged**